src/ast/nodes/ExpressionStatement.ts

```diff
@@ -2,6 +2,26 @@ import Statement from './shared/Statement';
 import MagicString from 'magic-string';
 
 export default class ExpressionStatement extends Statement {
+	directive?: string;
+
+	shouldBeIncluded() {
+		if (this.directive && this.directive !== 'use strict') {
+			if (this.parent.type === "Program") {
+				this.module.warn( // This is necessary, because either way (deleting or not) can lead to errors.
```

Comment:
`shouldBeIncluded` is run for each tree-shaking pass, which means the error is displayed at least twice, possibly more often. The rest of the code looks nice and I like the approach but the proper place to display the warning is to add a member `initialiseNode`, cf. e.g. the warning in `ThisExpression`. `initialiseNode` is run only once and is meant for node specific initialisations that do not depend on the scope being populated.